metric_learn/itml.py

```diff
@@ -151,8 +161,12 @@ def fit(self, pairs, y, bounds=None):
         preprocessor.
     y: array-like, of shape (n_constraints,)
         Labels of constraints. Should be -1 for dissimilar pair, 1 for similar.
-    bounds : list (pos,neg) pairs, optional
-        bounds on similarity, s.t. d(X[a],X[b]) < pos and d(X[c],X[d]) > neg
+    bounds : `list` of two numbers
+        Bounds on similarity, aside slack variables, s.t. ``d(a, b) < pos`` for
+        all given pairs of similar points ``a`` and ``b``, and
+        ``d(c, d) > neg`` for all given pairs of dissimilar points ``c`` and
+        ``d``, with ``bounds=[pos, neg]``, and ``d`` the learned distance.
+        If not provided at initialization, these will be derived at train time.
```

Comment:
I also updated the description of the argument in `fit` (`bounds`), to match the one of `self.bounds_`, and to be more detailed. Same comment: feel free to tell me if I'm wrong